### ISSUE TITLE
Stuff I would like to have available

### ISSUE THREAD

**Issue by VolvoBirkenstock** [mocaccinoOS/community-repository#28]
If you can tell me which packages contain these items, I would be more than happy to install them.  
Thank you very much in advance.  

gimp (Gnu Image Manipulation Program)
xemacs
gnu octave
gnuplot
thunderbird (or similar)
opera (I've got it on Sabayon but don't know if I really need it)
deluge (or similar)
gkrellm
brasero
k3b
gnome-disks
gparted
oracle vm virtualbox
as many games as I can find...got to entertain myself and others. ;-)

**Comment by msdobrescu** [comment]
Hi, if you know what you need, try `luet search` and the name of the app, like this, as `su`:

`luet search gimp`

or simply:

`sudo luet search gimp`

The output is like this:

```
 INFO   --- Search results (gimp): ---                                                                                                                                                                                                               
> apps/gimp-2.10.32
 -> Category: apps
 -> Name: gimp
 -> Version: 2.10.32
 -> Description: GNU Image Manipulation Program
 -> Repository: mocaccino-community-stable 
 -> Uri: https://www.gimp.org/ 
 -> Installed: false 
```

So there is a package for gimp (in the community repository which you must have it installed).

So, you issue as `su` or by using `sudo`:

`sudo luet install apps/gimp` and follow the steps (you will be asked to confirm).

Alternately, to skip confirmation, you can issue directly:

`sudo luet install apps/gimp -y`

Note, you need the `category` and the `name`, not the version, but it accepts the version if you provide it.

If you don't have community repository installed, you can perform a similar procedure for it:

`sudo luet search community`

and you get:
```

 INFO   --- Search results (community): ---                                                                                                                                                                                                                 
> repo-updater/mocaccino-community-stable-0.1+30
 -> Category: repo-updater
 -> Name: mocaccino-community-stable
 -> Version: 0.1+30
 -> Description: 
 -> Repository: mocaccino-repository-index 
 -> Uri:  
 -> Installed: false 
> repository/mocaccino-community-stable-20220912225446
 -> Category: repository
 -> Name: mocaccino-community-stable
 -> Version: 20220912225446
 -> Description: 
 -> Repository: mocaccino-repository-index 
 -> Uri:  
 -> Installed: true 
> repository/mocaccino-community-20220912225446
 -> Category: repository
 -> Name: mocaccino-community
 -> Version: 20220912225446
 -> Description: 
 -> Repository: mocaccino-repository-index 
 -> Uri:  
 -> Installed: false 
```

Above you see both repos,
- staging: `repository/mocaccino-community`
- stable: `repository/mocaccino-community-stable`

As end user, you install the stable one:

`sudo luet install repository/mocaccino-community-stable`

Note, in the above list you have the information stating if some package or repo is already installed:

` -> Installed: true`

Also note`luet search` supports partial names, as you can see above. In facts it supports a `regex`.

```
sudo luet search "gimp|xemacs"
 INFO   --- Search results (gimp|xemacs): ---                                                                                                                                                                                                               
> apps/gimp-2.10.32
 -> Category: apps
 -> Name: gimp
 -> Version: 2.10.32
 -> Description: GNU Image Manipulation Program
 -> Repository: mocaccino-community-stable 
 -> Uri: https://www.gimp.org/ 
 -> Installed: false 
```
The above means `gimp` is present in the repos, but not installed. There is no `xemacs`. Actually, there is `emacs` only.

Taking your list, I would do:

`sudo luet search "gimp|emacs|octave|gnuplot|thunderbird|opera|deluge|gkrellm|brasero|k3b|gnome-disks|gparted|virtualbox"`

**Comment by VolvoBirkenstock** [comment]
@msdobrescu Thank you!  Your response is excellent.  This is _exactly_ what I was looking for.  I'm not a developer and have not enough time to get in-depth on luet and mocaccino packages, so your nearly-cookbook instructions are exactly what I needed to read.  I am a little curious how you learned all this, but I am extremely appreciative that you shared your knowledge.

**Comment by VolvoBirkenstock** [comment]
I just thought of another question regarding luet and installation.  Can you install multiple applications with a single call to luet?  That is, perhaps something like "sudo luet install apps/gimp apps/emacs -y"?  Or can you install only one application with a single luet command?  (I haven't tried this and instead wrote a shellscript that runs many "luet install \<insertNameHere\> -y" commands.) 

Thank you in advance!

**Comment by msdobrescu** [comment]
Hi, don't mention it! I have learned `luet` directly from its creator, @mudler, and @joostruis. They dedicated a lot of time to teach me and I am glad to pass further that time and the knowledge :).

Indeed, `sudo luet install apps/gimp apps/emacs -y` works and It does not hurt the system to try it even though it wouldn't work.
Generally, `luet` would bring the deps too, which are apps and libs. If you encounter some conflicts and can't install, please report. In that case must uninstall the apps and the deps one by one. It's a case should not happen unless some libs are moved between packages by us, which may happen sometimes.

Enjoy our distro ;)!

**Comment by msdobrescu** [comment]
Hi, sorry for the confusion! _Au contraire_, **it works to specify several packages like that**. I wanted to say if you are not sure and try it, would not harm, in the worst case you get an error.

So issue `sudo luet install apps/gimp apps/emacs -y`, it works!

**Comment by msdobrescu** [comment]
Concerning the packages meaning, I can't tell all the packages by heart. I can do, for example:

```
luet search fluid
 INFO   --- Search results (fluid): ---                                                                                                                                                                                                                     
> apps/fluidsynth-2.2.8
 -> Category: apps
 -> Name: fluidsynth
 -> Version: 2.2.8
 -> Description: Software real-time synthesizer based on the Soundfont 2 specifications
 -> Repository: mocaccino-community-stable 
 -> Uri: https://www.fluidsynth.org 
 -> Installed: true 
> libs/fluid-soundfont-3.1
 -> Category: libs
 -> Name: fluid-soundfont
 -> Version: 3.1
 -> Description: The Fluid R3 soundfont
 -> Repository: mocaccino-community-stable 
 -> Uri: http://musescore.org/en/handbook/soundfont 
 -> Installed: true 
> libs/fluidsynth-dssi-1.0.0-r1
 -> Category: libs
 -> Name: fluidsynth-dssi
 -> Version: 1.0.0-r1
 -> Description: DSSI Soft Synth Interface
 -> Repository: mocaccino-community-stable 
 -> Uri: http://dssi.sourceforge.net/ 
 -> Installed: true 
```
Above, you see a description and a URL for the application's upstream source, taken from one of `https://packages.gentoo.org`,  the official Gentoo overlay, where `fluidsynth` search leads eventually to `https://packages.gentoo.org/packages/media-sound/fluidsynth`, or `https://gpo.zugaina.org`, the official and third party Gentoo overlays, that similarly leads to `https://gpo.zugaina.org/media-sound/fluidsynth`.

**Comment by VolvoBirkenstock** [comment]
I tried to replicate the gkrellm system monitoring graphical tool found on Sabayon onto my Mocaccino system.  In the Sabayon repositories it existed within app-admin/gkrellm.  I note that in mocaccino there exists an entity/gkrellmd, which I installed.  However, I cannot find the executable gkrellm which places the system monitoring graphical tool onto my display.  Also, when I use `sudo luet search gkrellm` the resulting "Description:" field is empty.  Perhaps the entity/gkrellmd package (or app) is just the daemon-side of the complete tool.  I don't know.  I don't know how to drive up the graphical tool.  I performed `find / -iname "*gkrellm*" -print` and found several gkrellm-related files but no gkrellm executable:
/etc/entities/gkrellmd (a directory containing group and passwd subdirectories)
/usr/share/vim/vim90/syntax/gkrellmrc.vim
/usr/share/bash-completion/completions/gkrellm
/usr/share/bash-completion/completions/gkrellm2
and of course
/var/luet/db/repos/mocaccino-os-commons-stable/treefs/entity/gkrellmd

And of course, being someone who likes games, I am looking for those as well.  LSkat, KPatience, Tetravex, KNetWalk, Konquest, KBlackBox, KSquares, Mahjongg, etcetera, ad nauseum, ad infinitum. ;-)  I just don't know how to find them yet.

**Comment by msdobrescu** [comment]
Entities are the packages for the system groups and/or users that are required by a specific software to run. There are entities defined in preparation for the future apps because those are deployed at a lower level than the community packages themselves.
That being said, I'd say `gkrellm` is not yet available. Also the games you have listed above.

As a general rule, packages requests are made for each piece of software alone, meaning, if you need some application or several, for each of those a request should be done.

Depending of their nature, will be added or not, to a layer of the main distro or to the community repo, according to our man power available, if available, but also to the planned milestones or current priorities etc.. I hope you understand.

**Comment by VolvoBirkenstock** [comment]
I understand.  It's perfectly understandable.  Macaccino is a work in progress.  From what I have observed in my new installation of Mocaccino, I think the developers have done an excellent job!

**Comment by msdobrescu** [comment]
Thank you for appreciation, in the name of the entire team!

**Comment by msdobrescu** [comment]
@VolvoBirkenstock, `gkrellm` was added. However, it _may not_ be accessible right now unless you force repository updates:

`sudo luet repo update`

Then search for it and install. I have no experience with it, could you please provide feed-back for it?

If you want a full update too, you may use:

`sudo luet repo update && luet upgrade -y'

After repos are forcibly updated and, generally, a system upgrade, _may need_ to run `sudo etc-update` for configs changes merging.

If things moved across packages, it's possible to have upgraded successfully, but still some binaries be missing due to the order the upgrade packages were applied. So, it's  advised to run a sanity check:
`sudo luet oscheck --reinstall` or `sudo luet oscheck --reinstall -y`, to automatically apply the fixes.
A restart is needed too if the check reinstalled or removed some packages. Then run the command again until you get a `SUCCESS All good!` message.

**Comment by VolvoBirkenstock** [comment]
Thank you, @msdobrescu !   I followed your instructions literally.  gkrellm is running and it checks out.  (After installing Mocaccino, I untar'd my data into my user space, and when I ran gkrellm and opened the configuration by right-clicking on the gkrellm window frame, I saw my old settings.)  I suspect that a future improvement would include a package called "lm_sensors" that is either not currently available or is already incorporated within your gkrellm package.  Within Sabayon, there was a package called sys-apps/lm_sensors.  I'm not asking for it at this time, but I am thankful for the generosity of your time and effort.  Very much appreciated!
